src/types.d.ts

```diff
@@ -0,0 +1,148 @@
+import { App } from 'electron'
+import { LngLat } from 'mapbox-gl'
+import * as DC from 'deltachat-node'
+
+export interface Login {
+  addr: string;
+}
+
+export interface AppState {
+  logins: Array<Login>;
+  deltachat: {
+    configuring: boolean;
+    credentials: Credentials;
+    ready: boolean;
+  }
+  saved: LocalSettings;
+}
+
+export interface ExtendedApp extends App{
+  rc: any;
+  isQuitting: boolean;
+  ipcReady: boolean;
+  localeData?: any
+  state?: any;
+  saveState?: (arg: AppState) => void;
+  translate?: (key: string, substitutions: Array<string>, opts: any) => string;
+}
+
+export interface Credentials {
+  addr: string;
+  passwd?: string;
```

Comment:
this should be here should it?

Reply:
Yes it should, since sometimes the credentials are passed without password like in deltachat state, so the password is optional.

Reply:
and where is it used?